src/javascripts/ng-admin/Crud/field/maChoiceField.js

```diff
@@ -17,18 +17,13 @@ function maChoiceField($compile) {
                     var template = `
                         <ui-select ng-model="$parent.value" ng-required="v.required" id="{{ name }}" name="{{ name }}">
                             <ui-select-match allow-clear="{{ !v.required }}" placeholder="Filter values">{{ $select.selected.label }}</ui-select-match>
-                            <ui-select-choices repeat="item.value as item in getChoices(entry) | filter: {label: $select.search}">
+                            <ui-select-choices repeat="item.value as item in choices | filter: {label: $select.search}">
                                 {{ item.label }}
                             </ui-select-choices>
                         </ui-select>`;
 
-                    var choices;
-                    if (field.type() === 'reference' || field.type() === 'reference_many') {
-                        choices = scope.datastore().getChoices(field);
-                    } else {
-                        choices = field.choices();
-                    }
-                    scope.getChoices = typeof(choices) === 'function' ? choices : function() { return choices; };
+                    var choices = field.choices();
+                    scope.choices = typeof(choices) === 'function' ? choices(scope.entry) : choices;
```

Comment:
I'm surprised this works, I had to return a function because Angular complained about an infinite look. Needs further testing.